doc/guides/debugging/coredump.rst

```diff
@@ -0,0 +1,347 @@
+.. _coredump:
+
+Core Dump
+#########
+
+The core dump module enables dumping the CPU registers and memory content
+for offline debugging. This module is called when fatal error is
+encountered, and the data is printed or stored according to which backends
+are enabled.
+
+Configuration
+*************
+
+Configure this module using the following options.
+
+* ``DEBUG_COREDUMP``: enable the module.
+
+Here are the options to enable output backends for core dump:
+
+* ``DEBUG_COREDUMP_BACKEND_LOGGING``: use log module for core dump output.
+* ``DEBUG_COREDUMP_BACKEND_NULL``: fallback core dump backend if other
```

Comment:
what is the purpose of having it ? Wouldn't just be better not enable coredump in this case ?

Reply:
I had it at first thinking I would write a build only test for this. But now this is not really needed, so removing it...